libs/SmartStore/src/com/salesforce/androidsdk/smartstore/app/SmartStoreSDKManager.java

```diff
@@ -133,26 +132,14 @@ public static SmartStoreSDKManager getInstance() {
     }
 
     @Override
-    protected void cleanUp(Activity frontActivity, Account account) {
-        if (account != null) {
-            final UserAccount userAccount = getUserAccountManager().buildUserAccount(account);
-            if (userAccount != null && hasSmartStore(userAccount)) {
-                DBOpenHelper.deleteDatabase(getAppContext(), userAccount);
-            }
+    protected void cleanUp(UserAccount userAccount) {
+        if (userAccount != null && hasSmartStore(userAccount)) {
+            DBOpenHelper.deleteDatabase(getAppContext(), userAccount);
         } else {
             DBOpenHelper.deleteAllUserDatabases(getAppContext());
         }
 
-        /*
-         * Checks how many accounts are left that are authenticated. If only one
-         * account is left, this is the account that is being removed. In this
-         * case, we can safely reset all DBs.
-         */
-        final List<UserAccount> users = getUserAccountManager().getAuthenticatedUsers();
```

Comment:
Don't we still need this? There's something special about the last user in a multi-user scenario. I don't remember the details. Or maybe we cleaned it up along the way.

Reply:
1) If there is only one account left, wouldn't that be the one passed to the cleanUp(UserAccount) method?
2) If somehow null is passed, but there is one account left, then I need to handle in the cleanUp(Activity, Account) method. cleanUp(UserAccount) could be running in a separate process as a result of the CLEANUP_INTENT_ACTION broadcast, and the account manager might already have been updated by then.
Do you think cleanup(null) could happen?

Reply:
Maybe this was put in to handle logout from `Settings` at some point, because `Settings` didn't have a handle on the account that was being removed. Looking at the code, I don't see a way `null` would be passed in for account either. We should be good.